python_modules/dagster/dagster/_core/definitions/unresolved_asset_job_definition.py

```diff
@@ -138,18 +139,26 @@ def resolve(
         assets: Sequence["AssetsDefinition"],
         source_assets: Sequence["SourceAsset"],
         default_executor_def: Optional["ExecutorDefinition"] = None,
+        asset_selection_resolver: Optional["AssetSelectionResolver"] = None,
     ) -> "JobDefinition":
         """
         Resolve this UnresolvedAssetJobDefinition into a JobDefinition.
         """
+        # when resolving many AssetSelection objects at once (e.g. in a repository), more efficient
+        # to create the AssetSelectionResolver object a single time
+        if asset_selection_resolver is not None:
+            asset_selection = asset_selection_resolver.resolve(self.selection)
```

Comment:
It's confusing that an `AssetSelection` gets resolved to an... `asset_selection`. What do you think about naming the variable `asset_keys`?